pkg/gitops/deploykey/deploykey.go

```diff
@@ -0,0 +1,45 @@
+package deploykey
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/fluxcd/go-git-providers/github"
+	"github.com/fluxcd/go-git-providers/gitprovider"
+	"github.com/pkg/errors"
+	giturls "github.com/whilp/git-urls"
+)
+
+const githubTokenVariable = "GITHUB_TOKEN"
+
+// GetDeployKeyClient provides a DeployKeyClient for the repo at the given URL
+func GetDeployKeyClient(ctx context.Context, url string) (gitprovider.DeployKeyClient, error) {
+	repo, err := giturls.Parse(url)
```

Comment:
You can use https://pkg.go.dev/github.com/fluxcd/go-git-providers@v0.0.1/gitprovider?tab=doc#ParseOrgRepositoryURL for this I think

Reply:
At the moment it can't parse `git@github.com:` URLs, I wasn't sure if that's intentional.

Reply:
Will work on this in separate PR!